viper.go

```diff
@@ -519,19 +529,24 @@ func SetEnvPrefix(in string) { v.SetEnvPrefix(in) }
 
 func (v *Viper) SetEnvPrefix(in string) {
 	if in != "" {
-		v.envPrefix = in
+		v.envPrefix = in + "_"
```

Comment:
Why append the underscore here if there is a separate variable tracking whether it should be?

Reply:
The variable tracks whether the underscore is added by viper (legacy behaviour) or by user.
The rest of the code does not bother whether the envPrefix was modified by viper or not.